lib/DxilDia/DxcPixDxilDebugInfo.cpp

```diff
@@ -229,7 +229,24 @@ dxil_debug_info::DxcPixDxilSourceLocations::DxcPixDxilSourceLocations(
         auto* S = llvm::dyn_cast<llvm::DIScope>(DL.getScope());
         while (S != nullptr && !llvm::isa<llvm::DIFile>(S))
         {
-            S = S->getScope().resolve(EmptyMap);
+            if(auto Namespace = llvm::dyn_cast<llvm::DINamespace>(S))
+            {
+                // DINamespace has a getScope member (that hides DIScope's)
+                // that returns a DIScope directly, but if that namespace
+                // is at file-level scope, it will return nullptr.
+                if (auto * ContainingScope = Namespace->getScope())
+                {
+                    S = ContainingScope;
+                }
+                else 
+                {
+                    S = S->getFile();
+                }
+            } 
+            else 
+            {
+                 S = S->getScope().resolve(EmptyMap);
```

Comment:
Initially, I wondered if the abover retrieved scopes required a resolve(), but now I'm wondering why this one is necessary. With an empty map, it looks like it will just cast to a pointer of DIScope or else it will assert

Reply:
There appears to be no syntactically-allowed way to go from a DIScopeRef to a DIScope? (It's really weird, parenthetically, that DIScope::getScope returns a DIScopeRef but DINamespace::getScope and DIModule::getScope return a DIScope* directly. Seems like that was just a mistake in the original design/implementation.)